packages/compartment-mapper/src/import-hook.js

```diff
@@ -8,18 +8,27 @@
 
 // @ts-check
 
-/** @import {ImportHook} from 'ses' */
+/** @import {ImportHook, ThirdPartyStaticModuleInterface} from 'ses' */
+/** @import {ImportNowHook} from 'ses' */
 /** @import {StaticModuleType} from 'ses' */
 /** @import {RedirectStaticModuleInterface} from 'ses' */
+/** @import {MakeImportNowHookMakerOptions, ModuleDescriptor} from './types.js' */
 /** @import {ReadFn} from './types.js' */
+/** @import {SyncReadPowers} from './types.js' */
 /** @import {ReadPowers} from './types.js' */
 /** @import {HashFn} from './types.js' */
 /** @import {Sources} from './types.js' */
 /** @import {CompartmentDescriptor} from './types.js' */
 /** @import {ImportHookMaker} from './types.js' */
 /** @import {ExitModuleImportHook} from './types.js' */
+/** @import {SourceMapHook} from './types.js' */
+/** @import {ImportNowHookMaker} from './types.js' */
 
-import { attenuateModuleHook, enforceModulePolicy } from './policy.js';
+import {
+  attenuateModuleHook,
+  ATTENUATORS_COMPARTMENT,
```

Comment:
I was able to contain  `ATTENUATORS_COMPARTMENT` reference to policy for the most part and from my experience refactoring my policy wiring in the initial PR many times, striving to hide it leads to good outcomes.

It'd be great if import hook didn't need to understand what  attenuators compartment is and the compartment "just worked" with the existing implementation. I'm for changing how attenuators are loaded if it's tripping up the import hook implementation, but it could also be an indicator that importNowHook relies on something it shouldn't.

Reply:
this might want a new issue

Reply:
It's become offtopic